<HTML>
<HEAD>
<meta charset="UTF-8">
<title>RPCException - corda</title>
<link rel="stylesheet" href="../../../style.css">
</HEAD>
<BODY>
<a href="../../index.html">corda</a>&nbsp;/&nbsp;<a href="../index.html">net.corda.client.rpc</a>&nbsp;/&nbsp;<a href="./index.html">RPCException</a><br/>
<br/>
<h1>RPCException</h1>
<code><span class="keyword">open</span> <span class="keyword">class </span><span class="identifier">RPCException</span>&nbsp;<span class="symbol">:</span>&nbsp;<a href="../../net.corda.core/-corda-runtime-exception/index.html"><span class="identifier">CordaRuntimeException</span></a></code>
<p>Thrown to indicate a fatal error in the RPC system itself, as opposed to an error generated by the invoked method.</p>
<h3>Constructors</h3>
<table>
<tbody>
<tr>
<td>
<p><a href="-init-.html">&lt;init&gt;</a></p>
</td>
<td>
<code><span class="identifier">RPCException</span><span class="symbol">(</span><span class="identifier" id="net.corda.client.rpc.RPCException$<init>(kotlin.String)/msg">msg</span><span class="symbol">:</span>&nbsp;<a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-string/index.html"><span class="identifier">String</span></a><span class="symbol">)</span></code><code><span class="identifier">RPCException</span><span class="symbol">(</span><span class="identifier" id="net.corda.client.rpc.RPCException$<init>(kotlin.String, kotlin.Throwable)/message">message</span><span class="symbol">:</span>&nbsp;<a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-string/index.html"><span class="identifier">String</span></a><span class="symbol">?</span><span class="symbol">, </span><span class="identifier" id="net.corda.client.rpc.RPCException$<init>(kotlin.String, kotlin.Throwable)/cause">cause</span><span class="symbol">:</span>&nbsp;<a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-throwable/index.html"><span class="identifier">Throwable</span></a><span class="symbol">?</span><span class="symbol">)</span></code>
<p>Thrown to indicate a fatal error in the RPC system itself, as opposed to an error generated by the invoked method.</p>
</td>
</tr>
</tbody>
</table>
<h3>Inherited Properties</h3>
<table>
<tbody>
<tr>
<td>
<p><a href="../../net.corda.core/-corda-runtime-exception/cause.html">cause</a></p>
</td>
<td>
<code><span class="keyword">open</span> <span class="keyword">val </span><span class="identifier">cause</span><span class="symbol">: </span><a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-throwable/index.html"><span class="identifier">Throwable</span></a><span class="symbol">?</span></code></td>
</tr>
<tr>
<td>
<p><a href="../../net.corda.core/-corda-runtime-exception/message.html">message</a></p>
</td>
<td>
<code><span class="keyword">open</span> <span class="keyword">val </span><span class="identifier">message</span><span class="symbol">: </span><a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-string/index.html"><span class="identifier">String</span></a><span class="symbol">?</span></code></td>
</tr>
<tr>
<td>
<p><a href="../../net.corda.core/-corda-runtime-exception/original-exception-class-name.html">originalExceptionClassName</a></p>
</td>
<td>
<code><span class="keyword">open</span> <span class="keyword">var </span><span class="identifier">originalExceptionClassName</span><span class="symbol">: </span><a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-string/index.html"><span class="identifier">String</span></a><span class="symbol">?</span></code></td>
</tr>
<tr>
<td>
<p><a href="../../net.corda.core/-corda-runtime-exception/original-message.html">originalMessage</a></p>
</td>
<td>
<code><span class="keyword">open</span> <span class="keyword">val </span><span class="identifier">originalMessage</span><span class="symbol">: </span><a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-string/index.html"><span class="identifier">String</span></a><span class="symbol">?</span></code></td>
</tr>
</tbody>
</table>
<h3>Inherited Functions</h3>
<table>
<tbody>
<tr>
<td>
<p><a href="../../net.corda.core/-corda-runtime-exception/add-suppressed.html">addSuppressed</a></p>
</td>
<td>
<code><span class="keyword">open</span> <span class="keyword">fun </span><span class="identifier">addSuppressed</span><span class="symbol">(</span><span class="identifier" id="net.corda.core.CordaRuntimeException$addSuppressed(kotlin.Array((kotlin.Throwable)))/suppressed">suppressed</span><span class="symbol">:</span>&nbsp;<a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-array/index.html"><span class="identifier">Array</span></a><span class="symbol">&lt;</span><a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-throwable/index.html"><span class="identifier">Throwable</span></a><span class="symbol">&gt;</span><span class="symbol">)</span><span class="symbol">: </span><a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-unit/index.html"><span class="identifier">Unit</span></a></code></td>
</tr>
<tr>
<td>
<p><a href="../../net.corda.core/-corda-runtime-exception/equals.html">equals</a></p>
</td>
<td>
<code><span class="keyword">open</span> <span class="keyword">fun </span><span class="identifier">equals</span><span class="symbol">(</span><span class="identifier" id="net.corda.core.CordaRuntimeException$equals(kotlin.Any)/other">other</span><span class="symbol">:</span>&nbsp;<a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-any/index.html"><span class="identifier">Any</span></a><span class="symbol">?</span><span class="symbol">)</span><span class="symbol">: </span><a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-boolean/index.html"><span class="identifier">Boolean</span></a></code></td>
</tr>
<tr>
<td>
<p><a href="../../net.corda.core/-corda-runtime-exception/hash-code.html">hashCode</a></p>
</td>
<td>
<code><span class="keyword">open</span> <span class="keyword">fun </span><span class="identifier">hashCode</span><span class="symbol">(</span><span class="symbol">)</span><span class="symbol">: </span><a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-int/index.html"><span class="identifier">Int</span></a></code></td>
</tr>
<tr>
<td>
<p><a href="../../net.corda.core/-corda-runtime-exception/set-cause.html">setCause</a></p>
</td>
<td>
<code><span class="keyword">open</span> <span class="keyword">fun </span><span class="identifier">setCause</span><span class="symbol">(</span><span class="identifier" id="net.corda.core.CordaRuntimeException$setCause(kotlin.Throwable)/cause">cause</span><span class="symbol">:</span>&nbsp;<a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-throwable/index.html"><span class="identifier">Throwable</span></a><span class="symbol">?</span><span class="symbol">)</span><span class="symbol">: </span><a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-unit/index.html"><span class="identifier">Unit</span></a></code></td>
</tr>
<tr>
<td>
<p><a href="../../net.corda.core/-corda-runtime-exception/set-message.html">setMessage</a></p>
</td>
<td>
<code><span class="keyword">open</span> <span class="keyword">fun </span><span class="identifier">setMessage</span><span class="symbol">(</span><span class="identifier" id="net.corda.core.CordaRuntimeException$setMessage(kotlin.String)/message">message</span><span class="symbol">:</span>&nbsp;<a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-string/index.html"><span class="identifier">String</span></a><span class="symbol">?</span><span class="symbol">)</span><span class="symbol">: </span><a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-unit/index.html"><span class="identifier">Unit</span></a></code></td>
</tr>
</tbody>
</table>
<h3>Extension Functions</h3>
<table>
<tbody>
<tr>
<td>
<p><a href="../../net.corda.core.utilities/kotlin.-any/context-logger.html">contextLogger</a></p>
</td>
<td>
<code><span class="keyword">fun </span><a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-any/index.html"><span class="identifier">Any</span></a><span class="symbol">.</span><span class="identifier">contextLogger</span><span class="symbol">(</span><span class="symbol">)</span><span class="symbol">: </span><span class="identifier">&lt;ERROR CLASS&gt;</span></code>
<p>When called from a companion object, returns the logger for the enclosing class.</p>
</td>
</tr>
</tbody>
</table>
<h3>Inheritors</h3>
<table>
<tbody>
<tr>
<td>
<p><a href="../-connection-failure-exception/index.html">ConnectionFailureException</a></p>
</td>
<td>
<code><span class="keyword">open</span> <span class="keyword">class </span><span class="identifier">ConnectionFailureException</span>&nbsp;<span class="symbol">:</span>&nbsp;<a href="./index.html"><span class="identifier">RPCException</span></a></code>
<p>Signals that the underlying <a href="../-r-p-c-connection/index.html">RPCConnection</a> dropped.</p>
</td>
</tr>
<tr>
<td>
<p><a href="../../net.corda.client.rpc.reconnect/-could-not-start-flow-exception/index.html">CouldNotStartFlowException</a></p>
</td>
<td>
<code><span class="keyword">class </span><span class="identifier">CouldNotStartFlowException</span>&nbsp;<span class="symbol">:</span>&nbsp;<a href="./index.html"><span class="identifier">RPCException</span></a></code>
<p>Thrown when a flow start command died before receiving a <a href="../../net.corda.core.messaging/-flow-handle/index.html">net.corda.core.messaging.FlowHandle</a>.
On catching this exception, the typical behaviour is to run a "logical retry", meaning only retry the flow if the expected outcome did not occur.</p>
</td>
</tr>
<tr>
<td>
<p><a href="../-max-rpc-retry-exception/index.html">MaxRpcRetryException</a></p>
</td>
<td>
<code><span class="keyword">class </span><span class="identifier">MaxRpcRetryException</span>&nbsp;<span class="symbol">:</span>&nbsp;<a href="./index.html"><span class="identifier">RPCException</span></a></code>
<p>Thrown to indicate an RPC operation has been retried for the <a href="#">maxNumberOfRetries</a> unsuccessfully.</p>
</td>
</tr>
<tr>
<td>
<p><a href="../-unrecoverable-r-p-c-exception/index.html">UnrecoverableRPCException</a></p>
</td>
<td>
<code><span class="keyword">open</span> <span class="keyword">class </span><span class="identifier">UnrecoverableRPCException</span>&nbsp;<span class="symbol">:</span>&nbsp;<a href="./index.html"><span class="identifier">RPCException</span></a></code>
<p>Thrown to indicate a fatal error in the RPC system which cannot be recovered from and so needs some manual support.</p>
</td>
</tr>
</tbody>
</table>
</BODY>
</HTML>
